tests/src/Unit/Plugin/migrate/process/EntityRevisionGenerateTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\Tests\stanford_migrate\Unit\Plugin\migrate\process;
+
+use Drupal\Core\DependencyInjection\Container;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityReferenceSelection\SelectionPluginManagerInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\RevisionableInterface;
+use Drupal\stanford_migrate\Plugin\migrate\process\EntityGenerateNoLookup;
+use Drupal\migrate\MigrateExecutable;
+use Drupal\migrate\Plugin\MigrateDestinationInterface;
+use Drupal\migrate\Plugin\MigratePluginManager;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\Row;
+use Drupal\stanford_migrate\Plugin\migrate\process\EntityRevisionGenerate;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Class EntityRevisionGenerateTest.
+ *
+ * @group stanford_migrate
+ * @coversDefaultClass \Drupal\stanford_migrate\Plugin\migrate\process\EntityRevisionGenerate
+ */
+class EntityRevisionGenerateTest extends UnitTestCase {
+
+  /**
+   * Test the tranform returns an entity.
+   */
+  public function testTranform() {
+    $container = new Container();
```

Comment:
So this is interesting. I just went through writing tests for a process plugin as well, and I started doing it this way, mocking everything for a unit test, but I ended up turning it into a kernel test and feeding it an an array as an embedded data source, and made it go through the whole migration, and checked for success.  Which do you think is the better approach?  I found myself getting frustrated with having to mock half the framework, and the kernel test came out much shorter, in terms of lines of code.  But clearly this is a legit way of doing it as well...

Reply:
There's a fine line between speed of writing the test and running the test. I normally try to go with the `Unit` test and mock everything, especially for `Plugin` classes due to their simplicity. After I get far enough and am tired of the mock objects I will go with a `Kernel` test. However with all tests, any API connection should be mocked regardless. 
Luckily for this one I just copied another test in the repo and tweaked it. Once you get one `Unit` test, it can be alot easier for others. You can even create an `abstact` class with a much of mocking and then extend that class for each test.